src/atlas/interpolation/method/MethodFactory.cc

```diff
@@ -47,6 +48,9 @@ void force_link() {
             MethodBuilder<method::GridBoxAverage>();
             MethodBuilder<method::GridBoxMaximum>();
             MethodBuilder<method::CubedSphereBilinear>();
+#if ATLAS_HAVE_EIGEN
```

Comment:
You don't need preprocessor guard here. It should still self-register, but throw a runtime exception when used.

Reply:
Done. What sorcery is this?!